hub/demo/src/server/api/routers/hub.ts

```diff
@@ -415,6 +416,59 @@ export const hubRouter = createTRPCRouter({
       return true;
     }),
 
+  forkEntry: protectedProcedure
+    .input(
+      z.object({
+        modifications: z.object({
+          name: z.string(),
+          version: z.string().nullish(),
+          description: z.string().nullish(),
+        }),
+        namespace: z.string(),
+        name: z.string(),
+        version: z.string(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      const response = await fetch(`${env.ROUTER_URL}/registry/fork`, {
+        method: 'POST',
+        headers: {
+          Authorization: ctx.authorization,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          modifications: input.modifications,
+          entry_location: {
+            namespace: input.namespace,
+            name: input.name,
+            version: input.version,
+          },
+        }),
+      });
+
+      if (response.status === 409) {
+        throw new TRPCError({
+          code: 'CONFLICT',
+          message: 'Test asdf',
```

Comment:
Update?
Since you're finding the conflicting resource server side you could pass through the name and skip the client side lookup. Fine to just put a placeholder in here though since it's all written.

Reply:
That's a good point. This additional throw logic isn't really needed due to the client side validation catching 99% of the issues with better UX. I simplified things here: https://github.com/nearai/nearai/pull/653/commits/50c505e55b6fd4d77b49d7d6a6126ef5fd9146b3